crates/matrix-sdk-base/src/read_receipts.rs

```diff
@@ -655,4 +927,574 @@ mod tests {
+                        let receipt_event = EventBuilder::new().make_receipt_event_content([
+                            (
+                                owned_event_id!("$2"),
+                                receipt_type_1.clone(),
+                                user_id.to_owned(),
+                                receipt_thread_1.clone(),
+                            ),
+                            (
+                                owned_event_id!("$3"),
+                                receipt_type_2.clone(),
+                                user_id.to_owned(),
+                                receipt_thread_2.clone(),
+                            ),
+                            (
+                                owned_event_id!("$1"),
+                                receipt_type_1.clone(),
+                                user_id.to_owned(),
+                                receipt_thread_2.clone(),
+                            ),
+                        ]);
+
+                        // Receipt-only sync, no new events, all receipts referred to events that
+                        // are known.
+                        let mut read_receipts = RoomReadReceipts::default();
+                        assert!(compute_notifications(
+                            user_id,
+                            room_id,
+                            Some(&receipt_event),
+                            all_events.clone(),
+                            &[],
+                            &mut read_receipts,
+                        )
+                        .unwrap());
+
+                        // $4 and $5 are unread.
+                        assert_eq!(read_receipts.num_unread, 2);
+                        assert_eq!(read_receipts.num_mentions, 0);
+                        assert_eq!(read_receipts.num_notifications, 0);
+
+                        // Receipt-only sync, mix of old and new events, all receipts referred to
+                        // events that are known.
+                        let mut read_receipts = RoomReadReceipts::default();
+                        assert!(compute_notifications(
+                            user_id,
+                            room_id,
+                            Some(&receipt_event),
+                            head_events.clone(),
+                            &tail_events,
+                            &mut read_receipts,
+                        )
+                        .unwrap());
+
+                        // $4 and $5 are unread.
+                        assert_eq!(read_receipts.num_unread, 2);
+                        assert_eq!(read_receipts.num_mentions, 0);
+                        assert_eq!(read_receipts.num_notifications, 0);
+                    }
+                }
+            }
+        }
+    }
+
+    /// Updating the pending list will cause a change in the `RoomReadReceipts`
+    /// fields, thus the function must return true.
```

Comment:
Would this comment be more relevant if it were next to the `compute_notifications` invocation? (I think that is what it is referring to.)

Reply:
Yep, tweaked that comment and added a note in the relevant place in `compute_notifications` :+1: